### ISSUE TITLE
compilation error for arm (softfp)

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#148]
@namhyung I've got the following compilation error when I try to build it for arm arch.
```
  CC FPIC  libmcount/dynamic.op
  LINK     arch/arm/uftrace.o
arm-linux-gcc: error: unrecognized command line option '-mno-sse2'; did you mean '-fno-dse'?
make: *** [/home/honggyu/src/uftrace/git/arm/uftrace/libmcount/dynamic.op] Error 1
make: *** Waiting for unfinished jobs....
  CC FPIC  libtraceevent/event-parse.o
```

**Comment by honggyukim** [comment]
Hm.. I reset the base to before making such error and got another error. fb70735
```
  CC FPIC  libmcount/plthook-fast.op
  CC FPIC  libmcount/mcount-single.op
/tmp/ccvbOWZe.s: Assembler messages:
/tmp/ccvbOWZe.s:1760: Error: selected processor does not support `vstr %d7,[r3,#16]' in ARM mode
/tmp/ccvbOWZe.s:1798: Error: selected processor does not support `vstr %s0,[r3,#16]' in ARM mode
/tmp/ccvbOWZe.s:1813: Error: selected processor does not support `vstr %s1,[r3,#16]' in ARM mode
    ...
/tmp/ccvbOWZe.s:2128: Error: selected processor does not support `vstr %d6,[r3,#16]' in ARM mode
/tmp/ccvbOWZe.s:2529: Error: selected processor does not support `vstr %s0,[r5,#16]' in ARM mode
/tmp/ccvbOWZe.s:2553: Error: selected processor does not support `vstr %d0,[r5,#16]' in ARM mode
make: *** [/home/honggyu/src/uftrace/git/arm/uftrace/arch/arm/mcount-support.op] Error 1
make: *** Waiting for unfinished jobs....
  CC FPIC  libtraceevent/event-plugin.o

```

**Comment by namhyung** [comment]
First issue looks like due to using host compiler for checking dependencies.  Does following patch make any difference?

```
diff --git a/configure b/configure
index 4c943ed..0ec7675 100755
--- a/configure
+++ b/configure
@@ -100,6 +100,8 @@ output=${output:-${objdir}/.config}
 MAKEFLAGS=
 MAKEOVERRIDES=
 
+export CC CFLAGS LD LDFLAGS
+
 make -siC ${srcdir}/check-deps check-clean
 make -siC ${srcdir}/check-deps check-build
 
```

**Comment by namhyung** [comment]
Second issue seems because the CPU (or toolchain?) doesn't support hardware floating-point.  We don't support those configuration (at least for now). Sorry.

**Comment by honggyukim** [comment]
@namhyung I successfully compiled it in a workaround way. Thanks for your help!

**Comment by namhyung** [comment]
OK did you succeed by commenting out the inline assembly?  Does it require additional changes?

Maybe we can add a preprocessor guard with config-time check.

**Comment by honggyukim** [comment]
OK did you succeed by commenting out the inline assembly? Does it require additional changes?

> I just commented out `vstr` related code.

Maybe we can add a preprocessor guard with config-time check.

> I agree. We can do it later.

**Comment by namhyung** [comment]
I don't think it's a bug - it's just an unsupported architecture variant.  But it might be better to (partially) support it by allowing to be compiled as you did.

**Comment by namhyung** [comment]
Could you please check [review/arm-softfp-v1](https://github.com/namhyung/uftrace/compare/review/arm-softfp-v1)?

**Comment by honggyukim** [comment]
Sorry but I'm not ready to test this as of now. Can I check it a bit later?

**Comment by namhyung** [comment]
Sure, take your time.  No need to rush for it. :)

**Comment by namhyung** [comment]
@kuzminrobin could you confirm that this makes the code to be compiled on a softfp environment?  Or did you need additional changes?

**Comment by kuzminrobin** [comment]
I would love to but unfortunately I'm extremely busy recently.
In general my epopee of compiling for ARM was so long and cumbersome that the full description will take about 10 screen pages. I hope within the next month I'll find the time to describe all that and share all my experience and advice.
I eventually got the `uftrace` binary compiling but the binary that was compiling right after `uftrace` failed to compile for me. I.e. I failed to reach the end of compile. I currently live with that.
